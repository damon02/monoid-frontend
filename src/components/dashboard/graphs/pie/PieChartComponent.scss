@import '../../../../style/style.scss';

.pietable {
  display: flex;
  padding: $spaces-small;
  background-color: $color-white;
}