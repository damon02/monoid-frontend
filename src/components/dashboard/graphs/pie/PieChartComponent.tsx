import * as React from 'react'
import { Pie, PieChart, ResponsiveContainer } from 'recharts'  
import { IGraphComponentData } from '../../../../statics/types'

interface IPieComponentProps {
  dataSet: IGraphComponentData[]
  dimensions?: { width: number, height: number }
  responsive?: { width: '100%', height: number } | { width: number, height: '100%' }
  showLabel?: boolean
}

export default class PieChartComponent extends React.PureComponent<IPieComponentProps, {}> {
  constructor(props : IPieComponentProps) {
    super(props)
  }

  public render() {
    const { dataSet, responsive, dimensions } = this.props
    const pies = dataSet.map((data, key) => {
      return (
        <Pie 
          key={key} 
          data={data.data} 
          dataKey={data.dataKey} 
          nameKey={data.nameKey} 
          label={data.label} 
          fill={data.color}
          cx={'50%'}
          cy={'50%'}
        />
      )}
    )

    return responsive ? (
        <ResponsiveContainer minHeight={250} minWidth={250} width={responsive.width} height={responsive.height}>
          <PieChart>
            {pies}
          </PieChart>
        </ResponsiveContainer>
      )
    : dimensions && (
      <PieChart width={dimensions.width} height={dimensions.height}>
        {pies}
      </PieChart>
    )
  }
}