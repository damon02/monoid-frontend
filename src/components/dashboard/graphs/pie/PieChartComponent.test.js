import React from 'react'
import ReactDOM from 'react-dom'
import PieChartComponent from './PieChartComponent'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

function data(){
  return [
    {
      color: '#32b4f1',
      dataKey: 'value',
      nameKey: 'name',
      data: [{ name: 'POST', value: 78 }, { name: 'GET', value: 271 }],
      label: true
    }
  ]
}

it('piechart renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <PieChartComponent dataSet={data()} responsive={{ width: 250, height: '100%' }}/>, 
    div
  )
})

describe('Piechart', () => {
  test('renders properly', () => {
    const tree = Renderer.create(
      <PieChartComponent dataSet={data()} responsive={{ width: 250, height: '100%' }} />,
      { createNodeMock }
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// it('piechart renders correctly', () => {
//   const tree = Renderer
//     .create(<PieChartComponent dataSet={data()} responsive={{ width: 250, height: '100%' }} />)
//     .toJSON()

//   expect(tree).toMatchSnapshot()
// })


