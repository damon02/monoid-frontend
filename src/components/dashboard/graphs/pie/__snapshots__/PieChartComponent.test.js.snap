// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piechart renders properly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "100%",
      "maxHeight": undefined,
      "minHeight": 250,
      "minWidth": 250,
      "width": 250,
    }
  }
>
  <div
    style={
      Object {
        "display": "none",
        "height": 0,
        "position": "absolute",
        "visibility": "hidden",
        "width": 0,
      }
    }
  />
</div>
`;
